### PR TITLE
Use NATIVE_DLL_SEARCH_DIRECTORIES to find libraries

### PR DESCRIPTION
**Pull request by morguldir**
NATIVE_DLL_SEARCH_DIRECTORIES is the same method to get paths that DllImport uses when loading libraries, this allows us to do what is suggested in #1638 without affecting this Native.Library.Load function.

- Complements #1638

---

Load libraries in the same way .NET does with the linux loading fix.

### CODE REVIEW

File: osu.Framework/Platform/Linux/Native/Library.cs [hunk 1]
```diff
@@ -14,9 +14,26 @@ public static class Library
         /// <summary>
         /// Loads a library with an enum of flags to use with dlopen. Uses <see cref="LoadFlags"/> for the flags
         /// </summary>
-        /// <param name="library">See 'man dlopen' for more information about the flags.</param>>
-        /// <param name="flags">See 'man ld.so' for more information about how the libraries are loaded.</param>>
-        public static void Load(string library, LoadFlags flags) => dlopen(library, flags);
+        /// <param name="flags">See 'man dlopen' for more information.</param>
+        /// <param name="locations">Uses ld.so and NATIVE_DLL_SEARCH_DIRECTORIES for library locations</param>
+        public static void Load(string library, LoadFlags flags)
```
Comment (VPeruS) [inline osu.Framework/Platform/Linux/Native/Library.cs]: method should look like:
```csharp
var paths = (string)AppContext.GetData("NATIVE_DLL_SEARCH_DIRECTORIES");
var pathList = paths.Split(':');
for (var path in pathList)
{
  if (dlopen(Path.Combine(path, library), flags) != IntPtr.Zero)
    break;
}
```

File: osu.Framework/Platform/Linux/Native/Library.cs [hunk 2]
```diff
@@ -14,9 +15,17 @@ public static class Library
...
-        /// <param name="flags">See 'man ld.so' for more information about how the libraries are loaded.</param>>
-        public static void Load(string library, LoadFlags flags) => dlopen(library, flags);
+        /// <param name="flags">See 'man dlopen' for more information.</param>
+        /// <param name="library">Uses ld.so and NATIVE_DLL_SEARCH_DIRECTORIES for locations</param>
+        public static void Load(string library, LoadFlags flags)
+        {
+            var paths = (string)AppContext.GetData("NATIVE_DLL_SEARCH_DIRECTORIES");
+            foreach (var path in paths.Split(':'))
+            {
+                if(dlopen(Path.Combine(path, library), flags) != IntPtr.Zero)
```
Comment (Tom94) [inline osu.Framework/Platform/Linux/Native/Library.cs]: Missing a space here. (Curious that CI does not complain)

File: osu.Framework/Platform/Linux/Native/Library.cs [hunk 3]
```diff
@@ -14,9 +15,17 @@ public static class Library
         /// <summary>
         /// Loads a library with an enum of flags to use with dlopen. Uses <see cref="LoadFlags"/> for the flags
         /// </summary>
-        /// <param name="library">See 'man dlopen' for more information about the flags.</param>>
-        /// <param name="flags">See 'man ld.so' for more information about how the libraries are loaded.</param>>
-        public static void Load(string library, LoadFlags flags) => dlopen(library, flags);
+        /// <param name="flags">See 'man dlopen' for more information.</param>
+        /// <param name="library">Uses ld.so and NATIVE_DLL_SEARCH_DIRECTORIES for locations</param>
```
Comment (VPeruS) [inline osu.Framework/Platform/Linux/Native/Library.cs]: >library full filename.<
and move it in summary

**Comment by VPeruS** [comment]
Tested with publish/run(with runtimes/) commands. Works as intended.

File: osu.Framework/Platform/Linux/Native/Library.cs [hunk 4]
```diff
@@ -12,11 +13,21 @@ public static class Library
...
         /// <summary>
-        /// Loads a library with an enum of flags to use with dlopen. Uses <see cref="LoadFlags"/> for the flags
+        /// Loads a library with flags to use with dlopen. Uses <see cref="LoadFlags"/> for the flags
+        ///
+        /// Uses NATIVE_DLL_SEARCH_DIRECTORIES and then ld.so for library paths
         /// </summary>
-        /// <param name="library">See 'man dlopen' for more information about the flags.</param>>
-        /// <param name="flags">See 'man ld.so' for more information about how the libraries are loaded.</param>>
-        public static void Load(string library, LoadFlags flags) => dlopen(library, flags);
+        /// <param name="flags">See 'man dlopen' for more information.</param>
```
Comment (peppy) [inline osu.Framework/Platform/Linux/Native/Library.cs]: param doc out of order

**Comment by peppy** [comment]
Let's get this in.

**Pull request merged**